CM.Text/BusinessMessaging/Model/MultiChannel/TextMessage.cs

```diff
@@ -28,11 +28,39 @@ public TextMessage(string text)
         }
 
         /// <summary>
-        /// A plain text message, when used it replaces the 'SMS' body text.
-        /// In <see cref="Channel.RCS"/>, when used in combination with an header and/or media this
-        /// will be set as the text of a rich card.
+        ///     A plain text message, when used it replaces the 'SMS' body text.
+        ///     In <see cref="Channel.RCS"/>, when used in combination with an header and/or media this will be set as the text of a rich card.
         /// </summary>
         [JsonProperty("text")]
         public string Text { get; set; }
+
+        /// <summary>
+        ///     Tag to send important and/or personally relevant 1:1 updates to recipients. E.g. to notify a recipient of an update on a recent purchase.
+        /// </summary>
+        [JsonProperty("tag")]
+        public string Tag { get; set; }
+
+        /// <summary>
+        ///     The suggestions of a text message.
+        /// </summary>
+        [JsonProperty("suggestions")]
+        public SuggestionBase[] Suggestions { get; set; }
+
+        /// <summary>
+        ///     Adds a suggestion
+        /// </summary>
+        /// <param name="suggestion"></param>
+        public void AddSuggestion(SuggestionBase suggestion)
+        {
+            if (this.Suggestions == null)
+                this.Suggestions = new[] { suggestion };
+            else
+            {
+                var newArr = this.Suggestions;
+                Array.Resize(ref newArr, this.Suggestions.Length + 1);
```

Comment:
This seems inefficient to me, is there a specific reason why `Suggestions` is an array instead of a collection you can later add objects to (like a list)? 
Or is it (mainly) due to the fact that it's a JSON property?

Reply:
No specific reason, it's copied from the RichContent model. Could of course made it better by making an extension out of it. But with the idea "Don't change it, if it works", I found it good enough to re-use it.

Reply:
Can one expect this function to be called for example 5 to 10 times, or only occasionally?
(otherwise, maybe a resize by doubling the size when needed might be an idea if it doesn't hurt to have the array be too large)

Another option would be to have a private backing field in the class that is a list, which gets converted to an array when the `Suggestions` property is gotten and set from array when set (meaning, by manually overwriting the get and set of the property). But only if this function can be expected to be used often

Reply:
TL;DR;
Change inbound. Going to remove this method.

Long story:
5 or 10 times is occasionally. Function will be called several times in one go for MT messages that must have a suggestion. So it will only be called for OTT channels, and only for exceptional MT messages. So it is really really occasional.

But I think it will be better to leave this method out of the model. Reason? Why add the possibility of adding one suggestion at a time to an array, while you probably know what suggestions you want to add before sending an MT message with suggestions. So an example of how it will be used most of times:

```
new TextMessage
{
   Suggestions = MethodThatReturnsIEnumerableOfSuggestions().ToArray()
}
```

Will leave the method in the RichContent-message to remain backwards compatible.

Reply:
That does indeed sound quite logical, and it seems like the best solution. Very well!